### PR TITLE
Include HABApp

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
I created the files but I am still a little bit lost since I've never written bash before.
How do I invoke the tests? In a docker container on my local machine?

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
````
Package python3-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python3-dev' has no installation candidate
E: Package 'python3-venv' has no installation candidate
FAILED
not ok 1 installation-habapp_install
````

The missing packages should actually be [available](https://packages.ubuntu.com/focal/python3-dev).
Do I need to add an apt update in the test to make it available?

**Comment by mstormi** [comment]
>  How do I invoke the tests? In a docker container on my local machine?

You can install & run BATS locally (probably easier while you're still working on getting tests to complete) or you can create a .bats file. All defined test then will be run on image build in a docker container in the cloud.
But for local BATS testing you don't need neither docker nor Python.

**Comment by spacemanspiff2007** [comment]
I have an application setup running when HABApp is starting.
What would be the best way to test if HABApp was installed on an openhabian machine?
Test if the ``/opt/openhabian`` folder exists?

**Comment by mstormi** [comment]
>  Test if the `/opt/openhabian` folder exists?

For sure not as it'll exist on any openHABian system.
Shouldn't you yourself know best how to test for your app?

**Comment by spacemanspiff2007** [comment]
The other way round. 
When the app is launched I want to check if it is running in the openhabian context so I can switch some defaults.

What would be the best way to update HABApp.
Since this is a python package it doesn't get updated through apt but through pypi.
Should there be a menu entry for it or should the install perform the update, too?

**Comment by mstormi** [comment]
>  When the app is launched I want to check if it is running in the openhabian context so I can switch some defaults.

I'd start it as a systemd service. Use can test for running services systemctl is-active <service>, see code for usage examples

>  Should there be a menu entry for it or should the install perform the update, too?

I hate those gazillion pkg managers all doing their own thing yum/apt, npm, pip, .... pypi? never even heard of.
openHABian is apt only and any external tools such as your need to take of its updates on its own.
A user should not need to care and not need to manually trigger updates as that's highly confusing him when we have a dozen tools running. So your app should take care of that yourself on startup.

**Comment by spacemanspiff2007** [comment]
Okay, then I guess this is it. Tests look good so far.

**Comment by mstormi** [comment]
Ask for a review when you think you're done.
Pls double-check all the info in CONTRIBUTING.md before you do.
And it'll take some days, we're more than busy.

**Comment by spacemanspiff2007** [comment]
I checked the install/uninstall multiple times on a fresh openhabian installation and it works as intended.
This can be reviewed (do I have to click something?)

**Comment by ecdye** [comment]
I don't think he can request because of the repo settings. @mstormi I will request a review from both of us as the active maintainers and will work on reviewing when I get a chance

File: functions/habapp.bash
Comment (mstormi) [inline functions/habapp.bash:46]: group openhabian preferred

File: functions/habapp.bash
Comment (mstormi) [inline functions/habapp.bash:10]: please use camelCase (i.e. start w/ lowercase) for variables

**Review by mstormi — COMMENTED** [review]
what about logging or other high disk activity ? Would we need to put this on ZRAM ?

mostly lgtm but need another round

**Comment by spacemanspiff2007** [comment]
> what about logging or other high disk activity ? Would we need to put this on ZRAM ?

The next release of HABApp will detect that it is on an openhabian machine and automatically log into the openhab-log folder.
The logfiles name contain HABApp, so it's clear that this is not an openhab log.
That way it should always work with ZRAM.

File: functions/habapp.bash
Comment (spacemanspiff2007) [reply functions/habapp.bash:46]: Is it openhabian:openhabian or openhabian:openhab?
Both have write access to the openhab-conf folder, right?

Comment (mstormi) [reply functions/habapp.bash:46]: openhab:openhabian
The standard "openhabian" or other admin Linux user isn't necessarily in the openhab group so may not have group access rights.
The openhab user has access anyway by user rights.

**Comment by spacemanspiff2007** [comment]
I have made the changes accordingly

**Comment by spacemanspiff2007** [comment]
I noticed that the other files in the conf folder are ``openhab:openhab`` and now the habapp folder is ``openhab:openhabian``.
Is this intended?

**Comment by mstormi** [comment]
>  Is this intended?

sort of, it's the oh default. On the habapp part: yes

**Comment by spacemanspiff2007** [comment]
Okay. I just verified it with the latest changes on a fresh image and it seems to run okay.

File: includes/habapp.service
Comment (ecdye) [inline includes/habapp.service:14]: Please add newline at the end of the file.

**Review by ecdye — CHANGES_REQUESTED** [review]
@spacemanspiff2007 Here are some of my thoughts on your code, it looks pretty good, just a few things need to be changed.

File: functions/habapp.bash
Comment (spacemanspiff2007) [reply functions/habapp.bash:21]: What do you mean?

Comment (ecdye) [reply functions/habapp.bash:21]: OK then

**Review by ecdye — APPROVED** [review]
LGTM, @mstormi once you approve, feel free to merge unless something large changes.

**Review by mstormi — APPROVED** [review]

**Comment by mstormi** [comment]
Thanks for your contribution. Let's hope people will make use of it.

**Comment by spacemanspiff2007** [comment]
I hope so, too. When will it be available?
Is it possible to use this on old OH2 installs, too?

**Comment by mstormi** [comment]
> When will it be available?

now

>  Is it possible to use this on old OH2 installs, too?

yes. I assumed you tested this ?

**Comment by spacemanspiff2007** [comment]
> yes. I assumed you tested this ?

Sure by manually updating the openhabian install.
I just wasn't sure if the openhabian gets updated independent of the openhab installation.

**Pull request merged**
